diff --git a/server/src/main/java/org/candlepin/service/impl/DefaultSubscriptionServiceAdapter.java b/server/src/main/java/org/candlepin/service/impl/DefaultSubscriptionServiceAdapter.java
index 51dd85b..67972dc 100644
--- a/server/src/main/java/org/candlepin/service/impl/DefaultSubscriptionServiceAdapter.java
+++ b/server/src/main/java/org/candlepin/service/impl/DefaultSubscriptionServiceAdapter.java
@@ -15,6 +15,7 @@
 package org.candlepin.service.impl;
 
 import org.candlepin.common.config.Configuration;
+import org.candlepin.common.exceptions.AcceptedRequestException;
 import org.candlepin.common.exceptions.ServiceUnavailableException;
 import org.candlepin.config.ConfigProperties;
 import org.candlepin.model.Consumer;
@@ -106,8 +107,37 @@ public class DefaultSubscriptionServiceAdapter implements
     @Override
     public void activateSubscription(Consumer consumer, String email,
         String emailLocale) {
-        throw new ServiceUnavailableException(
-            i18n.tr("Standalone candlepin does not support redeeming a subscription."));
+        String manufacturer = consumer.getFact("dmi.system.manufacturer");
+        String serialNumber = consumer.getFact("dmi.system.serial_number");
+
+        if (!manufacturer.toLowerCase().startsWith("dell")) {
+            throw new ServiceUnavailableException(
+                i18n.tr("Standalone candlepin does not support redeeming a subscription for dmi.system.manufacturer: {0}", manufacturer));
+        }
+        if (serialNumber.startsWith("1")) {
+            throw new AcceptedRequestException(
+                i18n.tr("The Dell service tag: {0}, has already been used to activate a subscription", serialNumber));
+        } else if (serialNumber.startsWith("2")) {
+             throw new AcceptedRequestException(
+                i18n.tr("A subscription was not found for the given Dell service tag: {0}", serialNumber));
+        } else if (serialNumber.startsWith("3")) {
+            throw new AcceptedRequestException(
+                i18n.tr("The Dell service tag: {0}, is expired", serialNumber));
+        } else if (serialNumber.startsWith("4")) {
+            throw new AcceptedRequestException(
+                i18n.tr("The system is unable to process the requested subscription activation {0}", serialNumber));
+        } else if (serialNumber.startsWith("5")) {
+            throw new AcceptedRequestException(
+                i18n.tr("Your subscription activation is being processed and should be available soon. " +
+                    "You will be notified via email once it is available. If you have any questions," +
+                    " additional information can be found here: https://access.redhat.com/kb/docs/DOC-53864.", serialNumber));
+        } else if (serialNumber.startsWith("6")) {
+            throw new AcceptedRequestException(
+                i18n.tr("The system is unable to process the requested subscription activation {0}", serialNumber));
+        } else {
+            throw new ServiceUnavailableException(
+                 i18n.tr("Standalone candlepin does not support redeeming a subscription for dmi.system.serial_number: {0}", serialNumber));
+        }
     }
 
     @Override
